packages/core/src/server.ts

```diff
@@ -1996,7 +2027,27 @@ export class Server {
         await Promise.all(promises);
 
         org.vaults = org.vaults.filter((v) => !deletedVaults.has(v.id));
-        org.members = org.members.filter((m) => !deletedMembers.has(m.id));
+        org.members = org.members.filter((m) => !m.accountId || !deletedMembers.has(m.accountId));
+    }
+
+    // Simplified version of the controller one
```

Comment:
@MaKleSoft This is the simplified version I created. Let me know if there's a better way you'd prefer (like injecting/exposing the `controller`).

Reply:
Actually, since we're not using this anywhere else, why not just implement this on `ScimServer`? The only important bit here is this:

```ts
auth = await this.storage.get(Auth, await getIdFromEmail(email));
```

and `storage` is already available to `ScimServer`.